sig-security-assessments/cluster-api/self-assessment.md

```diff
@@ -0,0 +1,1099 @@
+broad for CPI and CSI to function, and a management cluster hosted in AWS will
+have even more permissions, such as the ability to create and destroy EC2
+instances. A user of the cluster can start a pod, and scrape the IMDS endpoint
+living on link-local (`http://169.254.169.254`) and retrieve these credentials.
+They are short-lived but a confused deputy attack is still possible.
+
+Cloud Admin has access to multiple Cloud services like EC2, ELB, Secrets
+Manager. Access to the management cluster basically provides this admin access
+to all the managed workload clusters and its workloads as well.
+
+#### STRIDE-SPOOF-1 Recommended mitigations
+
+Securely manage Cloud and Kubernetes Credentials with following methods applied:
+
+* Credentials are short-lived
+  * Status: Implemented (AWS)
+* Credentials are managed centrally from an authoritative identity store
+  * Status: Implemented (AWS & Azure)
+* Credentials can be revoked, if needed
+  * Status: Implemented (AWS & Azure)
+* Second pair of eyes are applied when executing privileged actions such as
+  creating/deleting/updating a cluster
+  * Status: [End user guidance to be written](#cluster-api-end-user-guide)
+* Ensure credentials used by Cluster API are least privileges and setting access
+  control on Cluster API controller namespaces to prevent unauthorised access by
+  anyone other than cloud admin.
+  - Status: Implemented for K8s RBAC, developer guidance to be written for
+    providers
+* Enable auditing on the management cluster as well as on the cloud provider
+  * Status: [End user guidance to be written](#cluster-api-end-user-guide)
+
+#### STRIDE-SPOOF-2 Spoofing of cloud provider services
+
+Cloud provider services like Secrets Manager store sensitive data like cloudinit
+configuration files
+(shortlived). Other services like EC2 and ELB API have the ability to sniff
+traffic destined to the API server of the workload cluster. Successful spoofing
+of a cloud service could result in information disclosure and tampering of in
+flight requests relevant to cluster and machine management.
+
+#### STRIDE-SPOOF-2 Recommended Mitigations
+
+* Verify server side identity to confirm that AWS service is indeed what it
+  advertises to be.
+  * Status: Implemented (vSphere) / Out-of-scope (AWS/Azure)
+  * AWS maintainers note: We rely on
+    the [AWS Shared Responsibility model](#AWS-Shared-Responsibility-model).
+    There is realistically little we can do in the event of a major breach.
+    Validation of server endpoints is left to the standard SDK.
+
+#### STRIDE-SPOOF-3 Spoofing of image-builder maintainer
+
+Description: Machine image for cluster is pre-baked and built by cluster API
+maintainers. This image could be tampered with to include a cryptominer if the
+accounts of maintainers that have privileges and ownership over the code and
+machine image built for Nodes in a cluster.
+
+#### STRIDE-SPOOF-3 Recommended Mitigations
```

Comment:
As a CAPI project we cannot enforce how providers should manage their release process, but I like the idea of having security guidelines for providers in the CAPI book including this recommendation and others as well.
Would you be so kind to open an issue in CAPI with a list of recommendations we should surface?

Reply:
On second thoughts, we may need a separate issue that is tied to Providers only! Will work on opening this one soon.

Reply:
Added and linked